NotificationService/RLTNotificationsTracker.swift

```diff
@@ -19,19 +19,13 @@ public class RLTNotificationsTracker {
                            withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
         if let bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent),
            let userInfo = bestAttemptContent.userInfo as? [String: Any] {
-            configureSDKAndRegisterDevice { _ in
-                RealifeTech.Communicate?.trackPush(event: .received, trackInfo: userInfo) { _ in
-                    contentHandler(bestAttemptContent)
-                }
+            guard let sdkConfiguration = appGroupStore?.fetchSDKConfiguration() else {
+                return contentHandler(bestAttemptContent)
+            }
+            RealifeTech.configureSDK(with: sdkConfiguration)
+            RealifeTech.Communicate?.trackPush(event: .received, trackInfo: userInfo) { _ in
+                contentHandler(bestAttemptContent)
             }
         }
     }
-
-    private func configureSDKAndRegisterDevice(completion: @escaping (Bool) -> Void) {
-        guard let sdkConfiguration = appGroupStore?.fetchSDKConfiguration() else {
-            return completion(false)
-        }
-        RealifeTech.configureSDK(with: sdkConfiguration)
-        RealifeTech.General.registerDevice(completion)
```

Comment:
Previously the reason we need to call register device is because if we don't register SDK first, the`deviceRegistering.sdkReady` in AnalyticsLogger.swift will return false. which results in the PutAnalyticEvent mutation cannot be sent to BE.
So the fix here is to save the value of `sdkReady` in shared AppGroupUserDefault, so that once main app has already registered SDK, the notification extension target can fetch the value without calling the device register again.